app/src/main/java/com/stl/skipthelibrary/ViewableImage.java

```diff
@@ -2,43 +2,49 @@
 
 import android.graphics.Bitmap;
 import android.graphics.BitmapFactory;
-import android.net.Uri;
 import android.util.Base64;
 
 import java.io.ByteArrayOutputStream;
 import java.util.Objects;
 
 public class ViewableImage {
+
     private String imageString;
+
     
     public ViewableImage() {
         setImageString("default string");
     }
 
-    public ViewableImage(Uri uri) {
-        Bitmap bm = BitmapFactory.decodeFile(uri.toString());
+    public ViewableImage(Bitmap bitmap) {
+
         ByteArrayOutputStream baos = new ByteArrayOutputStream();
-        bm.compress(Bitmap.CompressFormat.JPEG, 100, baos); //bm is the bitmap object
+        bitmap.compress(Bitmap.CompressFormat.JPEG, 100, baos); //bm is the bitmap object
         byte[] b = baos.toByteArray();
+        encode(b);
```

Comment:
storing this guy on the user class makes the user pretty heavy it seems (memory) cause it was causing me issues with passing along the object as a parcel in an intent. Think we should look into compressing this? May be a thing to look into at a later point (could make an issue)

Reply:
We could have a separate images "table" in the database where either a user or a book is a "foreign key" then the images won't be passed around with the user?